tests/ssh/test_ssh_limit.py

```diff
@@ -0,0 +1,128 @@
+TEMPLATE_CREATE_COMMAND = "sudo touch {0}"
+
+PAM_LIMITS_TEMPLATE_PATH = "/usr/share/sonic/templates/pam_limits.j2"
+LIMITS_CONF_TEMPLATE_PATH = "/usr/share/sonic/templates/limits.conf.j2"
+LIMITS_CONF_TEMPLATE = "echo \"{{% if hwsku == '{0}' and type == '{1}'%}}\n{2}\n{{% endif %}}\"  | sudo tee {3}"
+
+def get_device_type(duthost):
+    device_config_db = json.loads(duthost.shell("sonic-cfggen -d --print-data")['stdout'])
+    dut_type = None
+
+    if "DEVICE_METADATA" in device_config_db and \
+        "localhost" in device_config_db["DEVICE_METADATA"] and \
+        "type" in device_config_db["DEVICE_METADATA"]["localhost"]:
+            dut_type = device_config_db["DEVICE_METADATA"]["localhost"]["type"]
+
+    return dut_type
+
+def modify_template(admin_session, template_path, additional_content, hwsku, type):
+    admin_session.exec_command(TEMPLATE_BACKUP_COMMAND.format(template_path))
+    admin_session.exec_command(TEMPLATE_CREATE_COMMAND.format(template_path))
+    admin_session.exec_command(LIMITS_CONF_TEMPLATE.format(hwsku, type, additional_content, template_path))
+
+def modify_templates(duthost, tacacs_creds, creds):
+    dut_ip = duthost.mgmt_ip
+    hwsku = duthost.facts["hwsku"]
+    type = get_device_type(duthost)
+    user = tacacs_creds['local_user']
+    
+    # Duthost shell not support run command with J2 template in command text.
+    admin_session = ssh_connect_remote(dut_ip, creds['sonicadmin_user'], creds['sonicadmin_password'])
+
+    # Backup and change /usr/share/sonic/templates/pam_limits.j2
+    additional_content = "session  required  pam_limits.so"
+    modify_template(admin_session, PAM_LIMITS_TEMPLATE_PATH, additional_content, hwsku, type)
+
+    # Backup and change /usr/share/sonic/templates/limits.conf.j2
+    additional_content = "{0}  hard  maxlogins  1".format(user)
+    modify_template(admin_session, LIMITS_CONF_TEMPLATE_PATH, additional_content, hwsku, type)
+
+def restore_templates(duthost):
+    duthost.shell(TEMPLATE_RESTORE_COMMAND.format(PAM_LIMITS_TEMPLATE_PATH))
+    duthost.shell(TEMPLATE_RESTORE_COMMAND.format(LIMITS_CONF_TEMPLATE_PATH))
+
+def restart_hostcfgd(duthost):
+    duthost.shell(HOSTSERVICE_RELOADING_COMMAND)
+    time.sleep(HOSTSERVICE_RELOADING_TIME)
+
+def limit_template_exist(duthost):
+    return duthost.stat(path=LIMITS_CONF_TEMPLATE_PATH).get('stat', {}).get('exists', False)
+
+@pytest.fixture(scope="module")
+def setup_limit(duthosts, rand_one_dut_hostname, tacacs_creds, creds):
+    duthost = duthosts[rand_one_dut_hostname]
+
+    # if template file not exist on duthost, ignore this UT
+    # However still need yield, if not yield, UT will failed with StopIteration error.
+    template_file_exist = limit_template_exist(duthost)
+    if template_file_exist:
+        setup_local_user(duthost, tacacs_creds)
+
+        # Modify templates and restart hostcfgd to render config files
+        modify_templates(duthost, tacacs_creds, creds)
```

Comment:
>modify_templates [](http://example.com/codeflow?start=8&length=16)

If the device already has a true `maxlogins` config, we do not need to modify it, and continue test with N+1 ssh sessions.